### PR TITLE
Fix NumRegField byte(s) read

### PR DESCRIPTION
**Pull request by prgeor**
#### Description
NumRegField read worked fine when the fields were single bit. Now that we have RegBitsFields with multiple bits we need to adjust the mask accordingly.

#### Motivation and Context
NumRegFields having RegBitsFields did not read the byte correctly. This is now fixed with this change.

#### How Has This Been Tested?
Unit test passed

### CODE REVIEW

**Comment by prgeor** [comment]
@andywongarista can you review this fix

File: sonic_platform_base/sonic_xcvr/fields/xcvr_field.py [hunk 1]
```diff
@@ -86,6 +87,7 @@ def __init__(self, name, bitpos, offset=None, **kwargs):
         super(RegBitField, self).__init__(name, offset, **kwargs)
         assert bitpos < 64
         self.bitpos = bitpos
+        self.bitmask = 1 << self.bitpos
```
Comment (AnoopKamath) [inline sonic_platform_base/sonic_xcvr/fields/xcvr_field.py:90]: Are we using bitmask in RegBitField ?

Comment (prgeor) [reply sonic_platform_base/sonic_xcvr/fields/xcvr_field.py:90]: @andywongarista not directly, but when RegBitField is grouped inside a NumberRegField or CodeRegField

Comment (AnoopKamath) [reply sonic_platform_base/sonic_xcvr/fields/xcvr_field.py:90]: @prgeor ok. Thanks

**Review by AnoopKamath — APPROVED** [review]

**Review by yxieca — APPROVED** [review]

**Pull request merged**